assets/js/modules/analytics-4/datastore/properties.test.js

```diff
@@ -1372,6 +1373,78 @@ describe( 'modules/analytics-4 properties', () => {
 				expect( fetchMock ).toHaveFetchedTimes( 1 );
 			} );
 
+			it( 'should cache the current property when fetched', async () => {
+				fetchMock.get( propertyEndpoint, {
+					body: fixtures.properties[ 0 ],
+					status: 200,
+				} );
+
+				const propertyID = '12345';
+				registry
+					.dispatch( MODULES_ANALYTICS_4 )
+					.setPropertyID( propertyID );
+
+				const initalPropertyCreateTime = registry
+					.select( MODULES_ANALYTICS_4 )
+					.getPropertyCreateTime();
+
+				expect( initalPropertyCreateTime ).toBeUndefined();
+
+				await untilResolved(
+					registry,
+					MODULES_ANALYTICS_4
+				).getPropertyCreateTime();
+				expect( fetchMock ).toHaveFetched( propertyEndpoint, {
+					query: { propertyID },
+				} );
+
+				const propertyCreateTimeInCache = await getItem(
+					`analytics4-properties-getPropertyCreateTime-${ propertyID }`
+				);
+
+				expect( propertyCreateTimeInCache.cacheHit ).toBe( true );
+				expect( propertyCreateTimeInCache.value ).toEqual(
+					fixtures.properties[ 0 ].createTime
+				);
+				expect( fetchMock ).toHaveFetchedTimes( 1 );
+			} );
+
+			it( 'should not make a request to the API if the property is cached', async () => {
+				const propertyID = fixtures.properties[ 0 ]._id;
+
+				const settings = {
+					propertyID,
+					webDataStreamID: '1000',
+					measurementID: 'abcd',
+					propertyCreateTime: 123456789,
+				};
+
+				await setItem(
+					`analytics4-properties-getPropertyCreateTime-${ propertyID }`,
+					settings
+				);
+
+				provideUserAuthentication( registry );
+
+				registry
+					.dispatch( MODULES_ANALYTICS_4 )
+					.receiveGetSettings( settings );
+
+				registry.select( MODULES_ANALYTICS_4 ).getPropertyCreateTime();
+				await untilResolved(
+					registry,
+					MODULES_ANALYTICS_4
+				).getPropertyCreateTime();
+
+				const propertyCreateTime = registry
+					.select( MODULES_ANALYTICS_4 )
+					.getPropertyCreateTime();
+				expect( propertyCreateTime ).toBe(
+					settings.propertyCreateTime
+				);
+				expect( fetchMock ).toHaveFetchedTimes( 0 );
+			} );
```

Comment:
I think the test case here is incorrect. The test passes even if the cache is not set, because `propertyCreateTime` is already being set in the `receiveGetSettings` dispatch.

Besides, the test case here sets the entire `settings` object in the cache, whereas in the resolver, we only store the `createTime`. See:

https://github.com/google/site-kit-wp/blob/e8330eef411d0a8696923bc73ce16e22747604a9/assets/js/modules/analytics-4/datastore/properties.js#L756-L762

```suggestion
			it( 'should not make a request to the API if the property is cached', async () => {
				const propertyID = fixtures.properties[ 0 ]._id;
				const expectedPropertyCreateTime = 123456789;

				const settings = {
					propertyID,
					webDataStreamID: '1000',
					measurementID: 'abcd',
				};

				await setItem(
					`analytics4-properties-getPropertyCreateTime-${ propertyID }`,
					expectedPropertyCreateTime
				);

				provideUserAuthentication( registry );

				registry
					.dispatch( MODULES_ANALYTICS_4 )
					.receiveGetSettings( settings );

				registry.select( MODULES_ANALYTICS_4 ).getPropertyCreateTime();
				await untilResolved(
					registry,
					MODULES_ANALYTICS_4
				).getPropertyCreateTime();

				const propertyCreateTime = registry
					.select( MODULES_ANALYTICS_4 )
					.getPropertyCreateTime();
				expect( propertyCreateTime ).toBe(
					expectedPropertyCreateTime
				);
				expect( fetchMock ).toHaveFetchedTimes( 0 );
			} );
```

Reply:
Could we also change the labels for these test cases so that it explicitly says only the createTime is cached, not the entire property? e.g.

Instead of `should cache the current property when fetched`, we can say `should cache the current property creation time when fetched`.

Thank you!

Reply:
Updated.